src/main/java/org/stellar/sdk/Transaction.java

```diff
@@ -248,11 +257,15 @@ public static Transaction fromV0EnvelopeXdr(TransactionV0Envelope envelope, Netw
 
   public static Transaction fromV1EnvelopeXdr(AccountConverter accountConverter, TransactionV1Envelope envelope, Network network) {
     int mFee = envelope.getTx().getFee().getUint32();
-    TimeBounds mTimeBounds = null;
     Long mSequenceNumber = envelope.getTx().getSeqNum().getSequenceNumber().getInt64();
     Memo mMemo = Memo.fromXdr(envelope.getTx().getMemo());
+    PreconditionsV2 preconditionsV2 = new PreconditionsV2();
+
     if (envelope.getTx().getCond().getDiscriminant().equals(PreconditionType.PRECOND_TIME)) {
-      mTimeBounds = TimeBounds.fromXdr(envelope.getTx().getCond().getTimeBounds());
+      preconditionsV2 = new PreconditionsV2.Builder().timeBounds(envelope.getTx().getCond().getTimeBounds()).build();
+    }
+    if (envelope.getTx().getCond().getDiscriminant().equals(PreconditionType.PRECOND_V2) && envelope.getTx().getCond().getV2().getTimeBounds() != null) {
+      preconditionsV2 = envelope.getTx().getCond().getV2();
```

Comment:
I can't tell if this is a naming conflict or incorrect functionality. If the envelope has PRECOND_TIME, we want to create PRECOND_TIME. If it has PRECOND_V2, we want to create PRECOND_V2 (regardless of whether or not those V2s actually have timebounds).

Reply:
this is unmarshalling tx xdr to java tx, so, once I refactor java tx.preconditions to be java pojo model instead of xdr then regardless of incoming tx xdr extension types, whichever attributes are available would converge/overlay into the one 'preconditions' pojo model here.